datafusion/src/optimizer/eliminate_limit.rs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Optimizer rule to replace `LIMIT 0` on a plan with an empty relation.
+//! This saves time in planning and executing the query.
+use crate::error::Result;
+use crate::logical_plan::LogicalPlan;
+use crate::optimizer::optimizer::OptimizerRule;
+
+use super::utils;
+
+/// Optimization rule that replaces LIMIT 0 with an [LogicalPlan::EmptyRelation]
+pub struct EliminateLimit {}
```

Comment:
Could this just be a unit struct, i.e. `pub struct EliminateLimit;` or is the intent to support optional future config in the struct? Not super-familiar with the optimiser setup.

Reply:
Nice suggestion! Wasn't even fully aware of that syntax, although I now do recognize it.

The idea to just have the same API as other rules, which are structured the same, and allows to implement the `OptimizerRule` trait. The optimizers can store some state or config in the struct when needed for the rule.